ARCs/arc-0065.md

```diff
@@ -0,0 +1,117 @@
+---
+arc: 65
+title: AVM Run Time Errors In Program
+description: Informative AVM run time errors based on program bytecode
+author: Cosimo Bassi (@cusma), Tasos Bitsios (@tasosbit), Steve Ferrigno (@nullun)
+discussions-to: https://github.com/algorandfoundation/ARCs/issues/315
+status: Draft
+type: Standards Track
+category: ARC
+created: 2024-10-09
+---
+
+## Abstract
+
+This document introduces conventions for rising informative run time error messages
+on the Algorand Virtual Machine (AVM) directly from the program bytecode.
+
+## Motivation
+
+The AVM does not offer native opcodes to catch and raise run time errors.
+
+The lack of native error handling semantics could lead to fragmentation of tooling
+and frictions for AVM clients, who are unable to retrieve informative and useful
+hints about the occurred run time failures.
+
+This ARC formalizes a convention to rise AVM run time errors based just on the program
+bytecode.
+
+## Specification
+
+The keywords "**MUST**", "**MUST NOT**", "**REQUIRED**", "**SHALL**", "**SHALL NOT**",
+"**SHOULD**", "**SHOULD NOT**", "**RECOMMENDED**", "**MAY**", and "**OPTIONAL**" 
+in this document are to be interpreted as described in <a href="https://datatracker.ietf.org/doc/html/rfc2119">RFC 2119</a>.
+
+> Notes like this are non-normative.
+
+### In Program Errors
+
+When a program wants to emit a run time error code, contained in the bytecode, 
+it **MUST**:
+
+1. Push to the stack the bytes string containing the error code;
+1. Push to the stack the `log` opcode;
+1. Push to the stack the `err` opcode.
```

Comment:
nit: these are opcodes that interact with the stack rather than being put on the stack.
```suggestion
1. Execute the `log` opcode to use the bytes from the top of the stack;
1. Execute the `err` opcode to immediately terminal the program.
```